File: 0187-Repeated DNA Sequences/0187-Repeated DNA Sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        vector<string> result;
        int n = s.size();
        if (n < 10)
        {
            return result;
        }
        unordered_map<string, int> table;
        for (int i = 0; i < n - 9; ++i)
        {
            string str = s.substr(i, 10);
            if (table[str]++ == 1)
            {
                result.push_back(str);
            }
        }
        
        return result;
    }
};

File: 0101-0200/0187-Repeated DNA Sequences/0187-Repeated DNA Sequences.py
import collections
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        table = collections.Counter()
        result = []
        for i in range(len(s) - 9):
            ss = s[i: i + 10]
            table[ss] += 1
            if table[ss] == 2:
                result.append(ss)
        return result
